BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/Skonto/SkontoViewModel.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  SkontoViewModel.swift
+//
+//  Copyright © 2024 Gini GmbH. All rights reserved.
+//
+
+import Foundation
+
+protocol SkontoViewModelDelegate: AnyObject {
+    func didTapHelp()
+    func didTapBack()
+    func didTapProceed(on viewModel: SkontoViewModel)
+}
+
+class SkontoViewModel {
+    private var skontoStateChangeHandlers: [() -> Void] = []
+    var endEditingAction: (() -> Void)?
+    var proceedAction: (() -> Void)?
+
+    private (set) var isSkontoApplied: Bool
+    private (set) var priceWithoutSkonto: Price
+    private (set) var priceWithSkonto: Price
+
+    var totalPrice: Price {
+        return isSkontoApplied ? priceWithSkonto : priceWithoutSkonto
+    }
+
+    private (set) var date: Date
+    private (set) var skontoValue: Double
+    private (set) var currencyCode: String
+
+    // TODO: recalculate with backend entity: skontoDuePeriod
+    var skontoFormattedDuePeriod: String {
+        return "14 days"
+    }
+
+    var skontoFormattedPercentageDiscounted: String {
+        let formatter = NumberFormatter()
+        formatter.minimumFractionDigits = 0
+        formatter.maximumFractionDigits = 2
+        formatter.numberStyle = .decimal
+        formatter.roundingMode = .floor
```

Comment:
[clarification] should we need to use here the NumberFormatter.localizedDecimal? Maybe we can make it even more generic

Reply:
NumberFormatter.localizedDecimal has some different settings
I will move this number formatter to extension to keep them both in one place

Reply:
I have added it to extension and a little bit optimised it to make less code duplicates. I also renamed them for more clear understanding